relay-server/src/actors/project_cache.rs

```diff
@@ -605,3 +617,44 @@ impl Handler<FlushBuckets> for ProjectCache {
         Box::new(future)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::{collections::HashMap, sync::Arc};
+
+    use relay_common::{ProjectId, ProjectKey};
+
+    use super::{Config, Project, ProjectCache, ProjectState};
+
+    #[test]
+    fn test_get_or_create_expired() {
+        let config = Arc::new(
+            Config::from_json_value(serde_json::json!(
+                {
+                    "cache": {
+                        "project_expiry": 0, // expire immediately
+                        "project_grace_period": 0,
+                        "eviction_interval": 9999 // do not evict
+                    }
+                }
+            ))
+            .unwrap(),
+        );
+
+        // Initialize project with a state
+        let project_key = ProjectKey::parse("a94ae32be2584e0bbd7a4cbb95971fee").unwrap();
+        let mut project_state = ProjectState::allowed();
+        project_state.project_id = Some(ProjectId::new(123));
+        let mut project = Project::new(project_key, config.clone());
+        project.set_state(project_state);
+        assert!(project.state().is_some());
+
+        // Initialize project cache with the created project:
+        let mut project_cache = ProjectCache::new(config, None);
+        project_cache.projects = HashMap::from([(project_key, project)]);
+
+        // Fetched project is expired, so it should not have state:
+        let project = project_cache.get_or_create_project(project_key);
+        assert!(project.state().is_none());
```

Comment:
If we introduce a regression where all projects are expired, this test will pass. Can we add another test case with a non-expired project state? That test will fail if the mentioned regression is introduced.